sdk/core/core-websockets/review/core-websockets.api.md

```diff
@@ -0,0 +1,102 @@
+## API Report File for "@azure/core-websockets"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import type { ClientRequestArgs } from 'node:http';
+import type * as WS from 'ws';
+
+// @public
+export function createWebSocketClient<WebSocketT>(url: string, options?: WebSocketClientOptions): WebsocketClientWrapper<WebSocketT>;
+
+// @public
+export type RetryMode = "Exponential" | "Fixed";
+
+// @public
+export interface RetryOptions {
+    maxRetries?: number;
+    maxRetryDelayInMs?: number;
+    mode?: RetryMode;
+    retryDelayInMs?: number;
+    timeoutInMs?: number;
+}
+
+// @public
+export interface WebSocketClient<WebSocketT> extends WebSocketEventRegistrar<WebSocketData> {
+    close(opts?: WebSocketCloseOptions): Promise<void>;
+    readonly identifier: string;
+    send(data: WebSocketData, opts?: WebSocketSendOptions): Promise<boolean>;
+    readonly status: WebSocketClientStatus;
+    readonly websocket: WebSocketT;
+}
+
+// @public
+export interface WebSocketClientOptions {
+    abortSignal?: AbortSignal;
+    allowInsecureConnection?: boolean;
+    highWaterMark?: number;
+    identifier?: string;
+    on?: Partial<WebSocketEventListeners<WebSocketData>>;
+    protocols?: string | string[];
+    retryOptions?: RetryOptions;
+    wsOptions?: WS.ClientOptions | ClientRequestArgs;
```

Comment:
Could we abstract this so we don't expose any types from WS?

Reply:
Good point, I put it under a ws subpath